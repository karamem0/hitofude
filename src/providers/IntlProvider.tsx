//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
