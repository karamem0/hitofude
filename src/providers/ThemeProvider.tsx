//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
