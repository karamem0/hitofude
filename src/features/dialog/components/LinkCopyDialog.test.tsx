//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './LinkCopyDialog.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import userEvent from '@testing-library/user-event';

jest.mock('../../../common/components/ModalDialog', () =>
  function ModalDialog({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="test-ModalDialog">
        {children}
      </div>
    );
  }
);

it('should create a shapshot when the loading copied is true', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    copied: true,
    mountNode: container,
    value: 'https://example.com'
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should create a shapshot when the loading copied is false', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    copied: false,
    mountNode: container,
    value: 'https://example.com'
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should raise onCopy event when click a button', async () => {
  const user = userEvent.setup();
  const container = document.body.appendChild(document.createElement('div'));
  const mock = jest.fn();
  const params = {
    copied: false,
    mountNode: container,
    value: 'https://example.com',
    onCopy: mock
  };
  render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  await user.click(screen.getByTitle('Copy'));
  expect(mock).toHaveBeenCalled();
});
