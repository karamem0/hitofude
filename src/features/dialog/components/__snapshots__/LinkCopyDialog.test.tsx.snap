// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a shapshot when the loading copied is false 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr3 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-ModalDialog"
    >
      <span
        hidden=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`should create a shapshot when the loading copied is true 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-ModalDialog"
    >
      <span
        hidden=""
      />
    </div>
  </div>
</DocumentFragment>
`;
